source/plugins/ruby/out_mdm.rb

```diff
@@ -108,11 +112,19 @@ def start
           @log.info "POST Request url: #{@@post_request_url}"
           ApplicationInsightsUtility.sendCustomEvent("AKSCustomMetricsMDMPluginStart", {})
 
-          # arc k8s cluster uses cluster identity
           if (!!@isArcK8sCluster)
-            @log.info "using cluster identity token since cluster is azure arc k8s cluster"
-            @cluster_identity = ArcK8sClusterIdentity.new
-            @cached_access_token = @cluster_identity.get_cluster_identity_token
+            if @isAADMSIAuth && !@isWindows
+              @log.info "using IMDS sidecar endpoint for MSI token since its Arc k8s and Linux node"
+              @useMsi = true
+              msi_endpoint = @@imds_msi_endpoint_template % { resource: @@token_resource_audience }
+              @parsed_token_uri = URI.parse(msi_endpoint)
+              @cached_access_token = get_access_token
+            else
```

Comment:
the else clause includes both: 

- @isAADMSIAuth && @isWindows 
- !@isAADMSIAuth && !@isWindows

is this intended?

Reply:
> the else clause includes both:
> 
> * @isAADMSIAuth && @isWindows
> * !@isAADMSIAuth && !@isWindows
> 
> is this intended?

> didn't understand, which line number you are referring.  Are you referring line#116?

Reply:
> > the else clause includes both:
> > 
> > * @isAADMSIAuth && @isWindows
> > * !@isAADMSIAuth && !@isWindows
> > 
> > is this intended?
>  IMDS side car endpoint only available  for Linux in  AAD MSI auth mode and for all other  possible scenarios, old way ( reading crd and secret) being used.